### PR TITLE
Fix no struct promotion condition for long vars on 32bit platforms.

### PR DESCRIPTION
**Pull request by sandreenko**
The condition was `(fgNoStructPromotion && varDsc->lvIsParam)` when it should be `fgNoStructPromotion || (fgNoStructParamPromotion && varDsc->lvIsParam))`. 

While I am here also move `lvaPromoteLongVars` to `DecomposeLongs` and add a comment `JitNoStructPromotion`.

There are tiny diffs in spmi tests on x86 from tests because we disable struct promotions in 2 cases:
https://github.com/dotnet/runtime/blob/b18be1e8c2f8f11303c25ce675cc3b0036747a8b/src/coreclr/jit/importer.cpp#L16474-L16476
and
https://github.com/dotnet/runtime/blob/b18be1e8c2f8f11303c25ce675cc3b0036747a8b/src/coreclr/jit/importer.cpp#L12130-L12135

and now, in these cases, we don't promote longs.

### CODE REVIEW

**Comment by sandreenko** [comment]
PTAL @dotnet/jit-contrib small cleaning change that came from another work item.

File: src/coreclr/jit/jitconfigvalues.h [hunk 1]
```diff
@@ -116,7 +116,8 @@ CONFIG_INTEGER(JitNoHoist, W("JitNoHoist"), 0)
 CONFIG_INTEGER(JitNoInline, W("JitNoInline"), 0)                 // Disables inlining of all methods
 CONFIG_INTEGER(JitNoMemoryBarriers, W("JitNoMemoryBarriers"), 0) // If 1, don't generate memory barriers
 CONFIG_INTEGER(JitNoRegLoc, W("JitNoRegLoc"), 0)
-CONFIG_INTEGER(JitNoStructPromotion, W("JitNoStructPromotion"), 0) // Disables struct promotion in Jit32
+CONFIG_INTEGER(JitNoStructPromotion, W("JitNoStructPromotion"), 0) // Disables struct promotion &1 - for all, &2 - for
```
Comment (kunalspathak) [inline src/coreclr/jit/jitconfigvalues.h:119]: nit: What does `&1` and `&2` mean?

**Review by kunalspathak — APPROVED** [review]
LGTM

File: src/coreclr/jit/jitconfigvalues.h (hunk 1 above)
Comment (sandreenko) [reply src/coreclr/jit/jitconfigvalues.h:119]: I do not know how to describe it, it means 
```
if ((JitNoStructPromotion & 1) != 0) fgNoStructPromotion = true
if ((JitNoStructPromotion & 2) != 0) fgNoStructParamPromotion= true
```

we use a similar "hash" approach for JitStress and JitMinOpts but they don't have such comments.

Comment (kunalspathak) [reply src/coreclr/jit/jitconfigvalues.h:119]: I see. Why not just pass `JitNoStructPromotion=1`, etc. and then check for `if (JitNoStructPromotion  == 1) fgNoStructPromotion = true` and so forth?

**Review by kunalspathak — APPROVED** [review]
Lgtm

**Pull request merged**